js/newSubmit.js

```diff
@@ -0,0 +1,1329 @@
+        cols: 72,
+        rows: 8
+    }), function() {
+        return {
+            valid: this.getValue().text.length >= Config.MIN_CHANGELOG_LENGTH,
+            message: "The changelog must be at least " + Config.MIN_CHANGELOG_LENGTH + " characters long"
+        };
+    }));
+
+    submitEntries.push("Help users find this plugin");
+    submitEntries.push(new SubmitFormEntry("majorCategory", submitData.fields.majorCategory, "Major Category", "submit2-majorcat", DroplistEntry(submitData.consts.categories, {}, function(newValue) {
+        var cbs = $("#submit2-minorcats").find(".cbinput");
+        cbs.removeClass("submit-cb-disabled").children(".submit-cb").prop("disabled", false);
+        var disabledCbInput = cbs.filter(function() {
+            var el = this.getElementsByClassName("submit-cb");
+            console.assert(el.length === 1);
+            return Number(el[0].value) === Number(newValue);
+        })
+            .addClass("submit-cb-disabled");
+        disabledCbInput.find(".submit-cb").prop("disabled", true);
+    }), function() {
+        return {valid: true};
+    }));
+    submitEntries.push(new SubmitFormEntry("minorCategories", submitData.fields.minorCategories, "Minor Categories", "submit2-minorcats", CompactMultiSelectEntry(submitData.consts.categories), function() {
+        return {valid: true};
+    }));
+    submitEntries.push(new SubmitFormEntry("keywords", submitData.fields.keywords, "Keywords", "submit2-keywords", StringEntry, function() {
+        var value = this.getValue();
+        if(value.split(" ").length > Config.MAX_KEYWORD_COUNT) {
+            return {
+                valid: false,
+                message: "Too many keywords! Only supply up to " + Config.MAX_KEYWORD_COUNT + " keywords."
+            };
+        }
+        for(var i = 0; i < value.length; ++i) {
+            if(value[i].length > Config.MAX_KEYWORD_LENGTH) {
+                return {
+                    valid: false,
+                    message: "The keyword \"" + value[i] + "\" is too long. Each keyword should not have more than " + Config.MAX_KEYWORD_LENGTH + " characters."
+                }
+            }
+        }
+        return {valid: true};
+    }));
+
+    submitEntries.push("About installation");
+    submitEntries.push(new SubmitFormEntry("deps", submitData.fields.deps, "Dependencies", "submit2-deps", DepTableEntry, function() {
+        return {valid: true};
+    }));
+    submitEntries.push(new SubmitFormEntry("requires", submitData.fields.reqrs, "Manual Setup", "submit2-requires", RequiresTableEntry, function() {
+        return {valid: true}
+    }));
+    submitEntries.push(new SubmitFormEntry("spoons", submitData.fields.spoons, "Supported APIs", "submit2-spoons", SpoonTableEntry, function() {
+        return {
+            valid: this.getValue().length >= 1,
```

Comment:
API rows should be locked while editing, because Poggit should not attempt to create a new phar/edit the phar after submitting.
I probably forgot to add that check in SubmitPluginAjax.